File: CodingChallengeSolutionsC++/38. Count and Say/38.CountandSay.cpp
#include <string>
class CountandSay {
public:
    std::string countAndSay(int n) {
        std::string result = "1";

        for (int itr = 1; itr < n; ++itr)
        {
            std::string newRLE = "";
            int counterLetters = 1;

            for (int x = 1; x < result.size(); ++x)
            {
                if (result[x] == result[x - 1])
                {
                    ++counterLetters;
                }
                else
                {
                    newRLE += std::to_string(counterLetters) + result[x - 1];
                    counterLetters = 1;
                }
            }
            newRLE += std::to_string(counterLetters) + result.back();
            result = newRLE;
        }

        return result;
    }
};

File: CodingChallengeSolutionsC++/38. Count and Say/38.CountandSay.py
class CountandSay:
    def countAndSay(self, n: int) -> str:
        result = "1"
        for _ in range(n - 1):
            result = self.describe(result)
        return result

    def describe(self, s: str) -> str:
        result = []
        counterLetters = 1

        for i in range(1, len(s)):
            if s[i] == s[i - 1]:
                counterLetters += 1
            else:
                result.append(f"{counterLetters}{s[i - 1]}")
                counterLetters = 1

        result.append(f"{counterLetters}{s[-1]}")
        return "".join(result)



